### PR TITLE
[iOS] Added doc on using UITableView as a hosting view for AdatpiveCard

### LINKED ISSUE #5650: [iOS][Document AdaptiveCard's usage with UITableView]
# Platform

What platform is your issue or question related to? (Delete other platforms).

- [ ] iOS

# Details

* There are some additional steps needed to be taken by the clients of AdaptiveCards if the clients are using UITableVew.
* The ask is to document these steps for reference.

### PR DESCRIPTION
**Pull request by jwoo-msft**
# Related Issue
Fixed #5650 

# Description
Added a doc on using UITableView as a hosting view for AdaptiveCard

# Sample Card

N/A

# How Verified
Made sure the documentation content is verified in the sample app.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/5738)

### CODE REVIEW

File: specs/DesignDiscussions/UITableView.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
+# UITableView
+## Problem Description
+Using UITableView as a hosting view for AdaptiveCard requires additional considerations. One of the challenges is the height changes in the row. UITableView correctly self sizes most of the times, but it breaks down in a few known cases, for example, if the images are loaded asynchronously causing height changes after the row is drawn, or if the visibility changes in its subviews, the row does not adjust the height dynamically. In such cases, there are a few well known practices in dealing with such occasions. This document covers practical guides in dealing with the height changes. This document will focus on [beginUpdates()](https://developer.apple.com/documentation/uikit/uitableview/1614908-beginupdates) and [reloadRows(at:with:)](https://developer.apple.com/documentation/uikit/uitableview/1614935-reloadrows) methods. It will show how to use the APIs in conjunction with AdaptiveCards' API. The rest of the documents uses the [iOS AdaptiveCard sample app](https://github.com/microsoft/AdaptiveCards/tree/main/source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizer) as an example.
```
Comment (jonmill) [inline specs/DesignDiscussions/UITableView.md:3]: `UITableView correctly self sizes most of the times, but it breaks down in a few known cases, for example, if the images are loaded asynchronously causing height changes after the row is drawn, or if the visibility changes in its subviews, the row does not adjust the height dynamically.`

This sentence is a bit difficult to read and super long...can you rephrase it a little bit?

File: specs/DesignDiscussions/UITableView.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+## Apple API
+1. [beginUpdates()](https://developer.apple.com/documentation/uikit/uitableview/1614908-beginupdates)
+     - As documented in the [Apple Documentation](https://developer.apple.com/documentation/uikit/uitableview/1614908-beginupdates), this method can be used to animate the changes in the row heights **wihtout reloading the cell** followed by endUpdates().
```
Comment (jonmill) [inline specs/DesignDiscussions/UITableView.md:6]: sp: `wihtout`

File: specs/DesignDiscussions/UITableView.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+## AdaptiveCard API
+The following API is used in conjunction with Apple API.
+1. `@protocol ACRActionDelegate <NSObject>`
+	* `- (void)didLoadElements` is a delegate that get called when AdaptiveCards' resources are all loaded.
+	* `- (void)didChangeVisibility:(UIButton *)button isVisible:(BOOL)isVisible` is a delegate when the visibility of the card is changed.
+	* `- (void)didChangeViewLayout:(CGRect)oldFrame newFrame:(CGRect)newFrame properties:(NSDictionary *)properties` is a delegate when card's layout has changed.
+
+## Refreshing the Layout Overview
+We want the delegate to be called only once after the card is rendered. Due to asynchronous tasks such as image loading, we have to do due diligence of ensuring the order of execution. In this example, we use GCD to ensure the right order. We create a block that renders an AdaptiveCard and queues it to the main queue. When the delegate is called, we wraps the layout refresh code in a block and queues it. This ensures that the layout refreshing happens after the rendering event.
```
Comment (jonmill) [inline specs/DesignDiscussions/UITableView.md:23]: sp: `wraps` -> wrap

**Review by jonmill — APPROVED** [review]
Just a couple of quick comments to address; it looks good to me otherwise! Great work!

**Review by golddove — APPROVED** [review]
Not sure this belongs in `DesignDiscussions`. As it applies specifically to one platform, maybe something like `guides/ios/`?

File: specs/DesignDiscussions/UITableView.md (hunk 1 above)
Comment (jwoo-msft) [reply specs/DesignDiscussions/UITableView.md:3]: sure; thank you for the feedback.

File: specs/DesignDiscussions/UITableView.md (hunk 2 above)
Comment (jwoo-msft) [reply specs/DesignDiscussions/UITableView.md:6]: thank you

File: specs/DesignDiscussions/UITableView.md (hunk 3 above)
Comment (jwoo-msft) [reply specs/DesignDiscussions/UITableView.md:23]: thank you

**Comment by jwoo-msft** [comment]
> Not sure this belongs in `DesignDiscussions`. As it applies specifically to one platform, maybe something like `guides/ios/`?

@golddove thanks risheek for your feedback. This will be definitely added to our doc site. I added it here because it is the design doc that I would have written for the protocol in the doc.

File: specs/DesignDiscussions/UITableView.md (hunk 3 above)
Comment (jwoo-msft) [reply specs/DesignDiscussions/UITableView.md:23]: thanks
